python-checks/src/main/java/org/sonar/python/checks/ItemOperationsTypeCheck.java

```diff
@@ -70,6 +69,48 @@ public boolean isValidSubscription(Expression subscriptionObject, String require
     return type.canHaveMember(requiredMethod);
   }
 
+  @CheckForNull
+  private static Boolean isValidSubscriptionSymbol(HasSymbol hasSymbol, Expression subscriptionObject, Map<LocationInFile, String> secondaries, String requiredMethod,
+    @Nullable String classRequiredMethod) {
+    Symbol symbol = hasSymbol.symbol();
+    if (symbol == null || isTypingOrCollectionsSymbol(symbol)) {
+      return true;
+    }
+    if (symbol.is(FUNCTION, CLASS)) {
+      secondaries.put(symbol.is(FUNCTION) ? ((FunctionSymbol) symbol).definitionLocation() : ((ClassSymbol) symbol).definitionLocation(),
```

Comment:
It will be easier to read if the location calculation will be extended and assigned to a local variable